docs/other-topics/hooks.md

```diff
@@ -2,404 +2,254 @@
+| `beforeBulkCreate`, `afterBulkCreate`                                                  | ✅     | When `Model.bulkCreate` is called                                               |
+| `beforeBulkDestroy`, `afterBulkDestroy`                                                | ✅     | When `Model.destroy` is called                                                  |
+| `beforeDestroy`, `afterDestroy`                                                        | ✅     | When `Model#destroy` is called                                                  |
+| `beforeBulkRestore`, `afterBulkRestore`                                                | ✅     | When `Model.restore` is called                                                  |
+| `beforeRestore`, `afterRestore`                                                        | ✅     | When `Model#restore` is called                                                  |
+| `beforeBulkUpdate`, `afterBulkUpdate`                                                  | ✅     | When `Model.update` is called                                                   |
+| `beforeUpdate`, `afterUpdate`                                                          | ✅     | When `Model#update` or `Model#save` is called (and the model isn't new)         |
+| `beforeCreate`, `afterCreate`                                                          | ✅     | When `Model#save` is called (and the model is new)                              |
+| `beforeSave`, `afterSave`                                                              | ✅     | When `Model#update` or `Model#save` is called                                   |
+
+### Registering Model hooks
+
+Instance Sequelize hooks can be registered in three ways:
+
+1. Using the `hooks` property of the `Sequelize` instance:
+   ```typescript
+   import { Sequelize, DataTypes } from '@sequelize/core';
+    
+   const sequelize = new Sequelize(/* options */);
+    
+   const MyModel = sequelize.define('MyModel', {
+     name: DataTypes.STRING,
+   });
+   
+   // highlight-next-line
+   MyModel.hooks.addListener('beforeFind', () => {
+     console.log('findAll has been called on MyModel');
+   });
+   ```
+2. Through the options of `Model.init`, or `sequelize.define`:
+   ```typescript
+   import { DataTypes } from '@sequelize/core';
+
+   const MyModel = sequelize.define('MyModel', {
+     name: DataTypes.STRING,
+   }, {
+     // highlight-next-line
+     hooks: {
+       beforeFind: () => {
+         console.log('findAll has been called on MyModel');
+       },
+     },
+   });
+   ```
+3. Using decorators
+   ```typescript
+   import { Sequelize, Model, Hook } from '@sequelize/core';
+   import { BeforeFindAll } from 'sequelize-typescript';
+   
+   export class MyModel extends Model {
+     // highlight-next-line
+     @BeforeFind
```

Comment:
This one could have remained `@BeforeFindAll` but if we are updating this, it might also be good to update the occurances later in this file

Reply:
It is `@BeforeFind`, there is no `@BeforeFindAll` in `sequelize-typescript`

Reply:
Ah, in that case you forgot a few instances of BeforeFindAll that should be replaced

Reply:
Damn, on it

Reply:
Done